pkg/epp/backend/provider.go

```diff
@@ -121,7 +121,8 @@ func (p *Provider) refreshMetricsOnce(logger logr.Logger) error {
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			updated, err := p.pmc.FetchMetrics(ctx, existing)
+			pool, _ := p.datastore.PoolGet()
```

Comment:
we need to check for the error here and skip updating if the pool wasn't set yet.